src/canvas/TracingPaths/TracingPath.tsx

```diff
@@ -68,21 +68,20 @@ export abstract class TracingPath implements ITracingPath {
         return true;
     };
 
-    snapStartToPoint = (point: Point): void => {
-        this._points[0] = point;
+    snapStartPoint = (paths: PatternPath[], point?: Point): boolean => {
+        const checkPoint = point || this._points[0];
+        return this._snapEndPoints(paths, checkPoint);
     };
 
-    snapEndToPoint = (point: Point): void => {
-        this._points[this._points.length -1] = point;
+    snapEndPoint = (paths: PatternPath[], point?: Point): boolean => {
```

Comment:
It looks like the method snapEndPoint doesn't ever reassign the last point of this._points. which is kinda what the name of the method promises it would do.

Reply:
Good catch! Thanks!